src/main/java/com/plotsquared/holoplots/DefaultHoloPlots.java

```diff
@@ -0,0 +1,204 @@
+package com.plotsquared.holoplots;
+
+import com.plotsquared.core.PlotSquared;
+import com.plotsquared.core.configuration.caption.Caption;
+import com.plotsquared.core.configuration.caption.CaptionMap;
+import com.plotsquared.core.configuration.caption.LocaleHolder;
+import com.plotsquared.core.configuration.caption.StaticCaption;
+import com.plotsquared.core.configuration.caption.TranslatableCaption;
+import com.plotsquared.core.database.DBFunc;
+import com.plotsquared.core.location.Location;
+import com.plotsquared.core.plot.Plot;
+import com.plotsquared.core.plot.PlotArea;
+import com.plotsquared.core.util.task.TaskManager;
+import com.plotsquared.holoplots.config.Configuration;
+import com.plotsquared.holoplots.provider.HologramProvider;
+import com.plotsquared.holoplots.provider.HologramProviderResolver;
+import io.papermc.lib.PaperLib;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import net.kyori.adventure.text.minimessage.tag.Tag;
+import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.World;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.SkullMeta;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class DefaultHoloPlots implements HoloPlots {
+
+    private static final String META_KEY_HOLOGRAM_UPDATE = "holoplots__pending_update";
+
+    private final HoloPlotsPlugin plugin;
+    private final HologramProvider provider;
+    private List<String> captions;
+
+    public DefaultHoloPlots(final HoloPlotsPlugin plugin, final HologramProviderResolver resolver) {
+        this.plugin = plugin;
+        this.provider = resolver.findProvider(this).orElseThrow(() -> {
+            Bukkit.getPluginManager().disablePlugin(plugin);
+            return new NullPointerException("Missing provider for spawning holograms. " +
+                    "Check https://github.com/IntellectualSites/HoloPlots for more information"
```

Comment:
If we cease Spigot support, we can link to modrinth's readme and add DH as dependency there :)